### ISSUE TITLE
Refit NPE Related to AmmoBins

### ISSUE THREAD

**Issue by mkerensky** [MegaMek/mekhq#571]
We've had a bunch of problems with these and fixed them, but here's yet one more. Customized a TBT-7M Trebuchet to one of the unofficial variants, changing LRM narc-capable to artemis-capable bins. There's an NPE now, and I can't access the repair bay tab to reload.   Log and savegame attached.
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/1583685/mekhqlog.txt)
[Botch.zip](https://github.com/MegaMek/mekhq/files/1583689/Botch.zip)

**Comment by HammerGS** [comment]
From the File
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at mekhq.campaign.parts.equipment.AmmoBin.isOmniPodded(AmmoBin.java:816)
	at mekhq.campaign.parts.equipment.AmmoBin.getActualTime(AmmoBin.java:453)
	at mekhq.campaign.parts.Part.getTimeLeft(Part.java:817)
	at mekhq.campaign.parts.equipment.AmmoBin.getDesc(AmmoBin.java:507)
	at mekhq.gui.model.TaskTableModel.getValueAt(TaskTableModel.java:52)
```

**Comment by neoancient** [comment]
The campaign file loads, but the log file shows that the equipmentNum field of the AmmoBin does not correspond to an equipment mount on the Unit's Entity. Added a null check would prevent the NPE but would likely hide an underlying problem. I suspect there's a logic bug in Utilities.unscrambleEquipmentNumbers.

**Comment by mkerensky** [comment]
Once I saved and loaded the game myself, I was able to reload the ammo bins from the repair bay.  My unit had 2 LRM-15 narc-capable ammo bins and went to a variant with 4 LRM15 artemis-capable bins.  I only had two in stock, though the refit told me I needed no parts, and it got left in a "2 and 2" state.  When I switched ammo in the two remaining narc bins through the hanger, that seemed to have an effect in clearing this up.
